src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,367 @@
+                                            double kx0, double dkx, double dkxy,
+                                            double ky0, double dky, double dkyx) const
+    {
+        dbg<<"SBInclinedExponential fillKValue\n";
+        dbg<<"kx = "<<kx0<<" + i * "<<dkx<<" + j * "<<dkxy<<std::endl;
+        dbg<<"ky = "<<ky0<<" + i * "<<dkyx<<" + j * "<<dky<<std::endl;
+        assert(val.stepi() == 1);
+        assert(val.canLinearize());
+        const int m = val.colsize();
+        const int n = val.rowsize();
+        typedef tmv::VIt<std::complex<double>,1,tmv::NonConj> It;
+
+        kx0 *= _r0;
+        dkx *= _r0;
+        dkxy *= _r0;
+        ky0 *= _r0_cosi;
+        dky *= _r0_cosi;
+        dkyx *= _r0_cosi;
+
+        It valit = val.linearView().begin();
+        for (int j=0;j<n;++j,kx0+=dkxy,ky0+=dky) {
+            double kx = kx0;
+            double ky = ky0;
+            for (int i=0;i<m;++i,kx+=dkx,ky+=dkyx) {
+                *valit++ = _flux * _info->kValue(kx,ky);
+            }
+        }
+    }
+
+    double SBInclinedExponential::SBInclinedExponentialImpl::maxK() const { return _info->maxK() * _inv_r0; }
+    double SBInclinedExponential::SBInclinedExponentialImpl::stepK() const { return _info->stepK() * _inv_r0; }
+
+    InclinedExponentialInfo::InclinedExponentialInfo(double h_tani_over_r, const GSParamsPtr& gsparams) :
+        _h_tani_over_r(h_tani_over_r),
+        _half_pi_h_tani_over_r(0.5*M_PI*h_tani_over_r),
+        _gsparams(gsparams),
+        _maxk(0.), _stepk(0.)
+    {
+        dbg<<"Start InclinedExponentialInfo constructor for h_tani_over_r = "<<h_tani_over_r<<std::endl;
+
+        assert(h_tani_over_r >= 0); // Should only ever have non-negative
+
+        // For large k, we clip the result of kValue to 0.
+        // We do this when the correct answer is less than kvalue_accuracy.
+        // (1+k^2 r0^2)^-1.5 = kvalue_accuracy
+        _ksq_max = (std::pow(gsparams->kvalue_accuracy,-1./1.5)-1.);
+
+        // For small k, we can use up to quartic in the taylor expansion to avoid the sqrt.
+        // This is acceptable when the next term is less than kvalue_accuracy.
+        // 35/16 (k^2 r0^2)^3 = kvalue_accuracy
+        _ksq_min = std::pow(gsparams->kvalue_accuracy * 16./35., 1./3.);
+
+        // Calculate stepk (assuming that this is similar enough to an exponential disk):
+        // int( exp(-r) r, r=0..R) = (1 - exp(-R) - Rexp(-R))
+        // Fraction excluded is thus (1+R) exp(-R)
+        // A fast solution to (1+R)exp(-R) = x:
+        // log(1+R) - R = log(x)
+        // R = log(1+R) - log(x)
+        double logx = std::log(gsparams->folding_threshold);
+        double R = -logx;
+        for (int i=0; i<3; i++) R = std::log(1.+R) - logx;
+        // Make sure it is at least 5 hlr
+        // half-light radius = 1.6783469900166605 * r0
+        const double hlr = 1.6783469900166605;
```

Comment:
I agree that this is probably conservative and fine.  Mostly, I think this needs an inline comment explaining that this hlr value is the value for Exponential, not the current profile, and that the real hlr is smaller, so this choice is conservative.  The current inline comment that you copied from the Exponential class is simply inaccurate in this context.  

Similarly, above you say "assuming that this is similar enough to an exponential disk", but really the salient point is that the flux enclosed in a radius R will be >= the flux from an exponential, so this is a conservative choice for that calculation as well.  

Both of these points should be explained in comments for the next person who comes along to look at this code.